### PR TITLE
Only enable HARDENED_MALLOC with kernel 6.1+

### PR DESCRIPTION
**Pull request by nextgens**
## What type of PR?

enhancement

## What does this PR do?

Disable HARDENED_MALLOC unless the requirements are met

### Related issue(s)
- #3398

## Prerequisites
Before we can consider review and merge, please make sure the following list is done and checked.
If an entry in not applicable, you can check it or remove it from the list.

- [ ] In case of feature or enhancement: documentation updated accordingly
- [x] Unless it's docs or a minor change: add [changelog](https://mailu.io/master/contributors/workflow.html#changelog) entry file.

### CODE REVIEW

File: core/base/libs/socrate/socrate/system.py [hunk 1]
```diff
@@ -58,6 +58,11 @@ def flush(self):
         self.stream.flush()
 
 def _is_compatible_with_hardened_malloc():
+    with open('/proc/version', 'r') as f:
```
Comment (DjVinnii) [inline core/base/libs/socrate/socrate/system.py:61]: It is also possible to get the kernel version from the[ `platform` module](https://docs.python.org/3/library/platform.html#platform.release).

Comment (DjVinnii) [reply core/base/libs/socrate/socrate/system.py:61]: Same goes for the checking the CPU features on line 66:

```python
import platform

cpu_info = platform.processor()

if 'avx2' not in cpu_info:
    return False
if 'lrcpc' not in cpu_info:
    return False
```

Comment (nextgens) [reply core/base/libs/socrate/socrate/system.py:61]:
```
root@a1-full:/mailu# docker compose exec admin python3
Python 3.12.3 (main, Apr 18 2024, 07:52:31) [GCC 13.2.1 20240309] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import platform
>>> 'lrcpc' in platform.processor()
False
>>> 
root@a1-full:/mailu# grep -c1 lrcpc /proc/cpuinfo 
4
```

Comment (DjVinnii) [reply core/base/libs/socrate/socrate/system.py:61]: Oh whoops. That is indeed not the case, my bad. However `platform.release()` will work for the kernel version.

**Review by Diman0 — APPROVED** [review]
lgtm

**Pull request merged**